### PR TITLE
Add loops to doc list of things not stable in const fn

### LINKED ISSUE #61508: Using `while` in `const fn` results in a misleading error message on stable
[Playground Link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=bfea3854b128266e330b3aac6df69010)

When `while` is used in a `const fn` on stable, it produces an error message: `` `if`, `match`, `&&` and `||` are not stable in const fn ``. This is a bit confusing because the function does not explicitly contain `if`, `match`, `&&`, or `||`.

I would expect the error message to also mention that `while` is disallowed, or say that "loops are not allowed in const fn" (which it does when `loop {}` construct is encountered).

(It seems like it would be even better to have a specific error message for the disallowed construct that the compiler encounters rather than just providing a list of disallowed constructs, if this is possible architecturally.)

<details>
<summary>rustc version</summary>

```
rustc 1.35.0 (3c235d560 2019-05-20)
binary: rustc
commit-hash: 3c235d5600393dfe6c36eeed34042efad8d4f26e
commit-date: 2019-05-20
host: x86_64-apple-darwin
release: 1.35.0
LLVM version: 8.0
```
</details>

### PR DESCRIPTION
**Pull request by czipperz**
Closes #61508

### CODE REVIEW

File: src/test/ui/consts/single_variant_match_ice.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ impl Foo {
         use self::Foo::*;
 
         match *self {
-            Prob => 0x1, //~ ERROR `if`, `match`, `&&` and `||` are not stable in const fn
+            Prob => 0x1, //~ ERROR `if`, `match`, `while`, `for`, `loop`, `&&` and `||` are not stable in const fn
```
Comment (zackmdavis) [inline src/test/ui/consts/single_variant_match_ice.rs]: move expectation comment to the next line and use the `//~^` notation get around tidy

File: src/librustc_mir/transform/qualify_min_const_fn.rs [hunk 2]
```diff
@@ -299,7 +299,7 @@ fn check_terminator(
 
         TerminatorKind::FalseEdges { .. } | TerminatorKind::SwitchInt { .. } => Err((
             span,
-            "`if`, `match`, `&&` and `||` are not stable in const fn".into(),
+            "`if`, `match`, `while`, `for`, `loop`, `&&` and `||` are not stable in const fn".into(),
```
Comment (zackmdavis) [inline src/librustc_mir/transform/qualify_min_const_fn.rs]: use `\` continuation to satisfy tidy?

**Review by zackmdavis — COMMENTED** [review]
CI is complaining about line length.

```
tidy check
[00:04:55] tidy error: /checkout/src/librustc_mir/transform/qualify_min_const_fn.rs:302: line longer than 100 chars
[00:04:55] tidy error: /checkout/src/test/ui/consts/single_variant_match_ice.rs:18: line longer than 100 chars
[00:04:59] some tidy checks failed
```

**Comment by czipperz** [comment]
@zackmdavis Thanks for the feedback.  Honestly I think this error message is too long.  What do you think about trying to reduce the length?  I'm trying to bike shed some new messages.  Maybe:

    conditional expressions are not stable in const fn

**Comment by Centril** [comment]
@czipperz I think:

> loops and conditional expressions are not stable in const fn

would do.

r? @Centril r=me with that

File: src/librustc_mir/transform/qualify_min_const_fn.rs [hunk 3]
```diff
@@ -299,7 +299,7 @@ fn check_terminator(
 
         TerminatorKind::FalseEdges { .. } | TerminatorKind::SwitchInt { .. } => Err((
             span,
-            "`if`, `match`, `&&` and `||` are not stable in const fn".into(),
+            "loops and conditional statements are not stable in const fn".into(),
```
Comment (Centril) [inline src/librustc_mir/transform/qualify_min_const_fn.rs]:
```suggestion
            "loops and conditional expressions are not stable in const fn".into(),
```

**Comment by czipperz** [comment]
@centril done

**Pull request merged**